### DISCUSSION TITLE
ViewModel to ViewModel communication

### DISCUSSION THREAD

**Discussion by leeleito** [Caliburn-Micro/Caliburn.Micro#840] [Q&A]
Hope this make sense!

I'm doing some refactoring on a large WPF application with approx 250 view models. At the moment we are creating child viewmodel withthin existing viewmodels  ParentViewModel > creates a new ChildModelA, ChildModelB etc then the ParentViewModel is directly calling methods on the child models - such as createnewmodel, fetchmodel

I now realise that using the eventaggregator and IHandle is probably the better way forward. I'd create a class on each viewmodel for the parameters.

Do I really need need to create a class for each viewmodel so each one handles the correct event from the parent and not get fired into every viewmodel? I've seen examples of adding a profile tag is this a better way forward?

**Comment by Yinimi** [comment]
It's probably a bit late to answer this, but anyway.
I do not recommend to use IHandle stuff as long as not necessarily needed. It makes the code hard to read and to refactor.
Why don't you use for example a service that can be injected and has a event where ViewModels can subscribe to?
